kafka-rest/src/main/java/io/confluent/kafkarest/response/StreamingResponse.java

```diff
@@ -255,7 +255,11 @@ public void close() {
       try {
         inputStream.close();
       } catch (IOException e) {
-        log.error("Error when closing the request stream.", e);
+        log.error("Error when closing the request stream", e);
+      } catch (BadRequestException e) {
```

Comment:
@akhileshchg thanks for fixing into this.

Can you add a comment here to explain why handling these exception are important w.r.t thread leaks?